main/src/cgeo/geocaching/calculator/ButtonData.java

```diff
@@ -0,0 +1,111 @@
+package cgeo.geocaching.calculator;
+
+import static cgeo.geocaching.models.CalcState.ERROR_CHAR;
+
+import java.io.Serializable;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Data used to capture the state of this particular button such that it can be restored again later
+ */
+public final class ButtonData implements Serializable, JSONAble {
```

Comment:
c:geo does not use Serializable put Parcelable. But also the later only if needed. I assume here it is probably not needed at all? (since you use json)

Reply:
Done

Reply:
Hm, I am not sure, if it's needed for storing in DB? But I guess, there only the JSON-String is stored.

On testing on the emulator, I got some leaks-messages, which I have not seen before. I will switch to local master-branch, to check, whether they appear there as well or I have to take a closer look at this. Probably I will need your help then ...

Reply:
I can assure you that serialization is not needed for db store. 

Was this also in the classes before? (In case you moved them) I have the strong feeling it is not needed, but it also doesn't hurt.

Reply:
It was already in the class before move.
I moved it then for VariableData as well.